code/game/objects/items/melee/f13powerfist.dm

```diff
@@ -156,31 +156,68 @@
 /obj/item/twohanded/chainsaw
 	name = "chainsaw"
 	desc = "A versatile power tool. Useful for limbing trees and delimbing humans."
-	icon_state = "chainsaw_off"
-	lefthand_file = 'icons/mob/inhands/weapons/chainsaw_lefthand.dmi'
-	righthand_file = 'icons/mob/inhands/weapons/chainsaw_righthand.dmi'
-	flags_1 = CONDUCT_1
-	force = 13
+	icon = 'icons/fallout/objects/melee/melee.dmi'
+	lefthand_file = 'icons/fallout/onmob/weapons/melee2h_lefthand.dmi'
+	righthand_file = 'icons/fallout/onmob/weapons/melee2h_righthand.dmi'
+	icon_state = "chainsaw"
+	item_state = "chainsaw"
+	var/on_icon_state = "chainsaw_on"
+	var/off_icon_state = "chainsaw"
+	var/on_item_state = "chainsaw_on"
+	var/off_item_state = "chainsaw"
+	w_class = WEIGHT_CLASS_NORMAL
+	var/weight_class_on = WEIGHT_CLASS_HUGE
+	total_mass = TOTAL_MASS_MEDIEVAL_WEAPON
+	slot_flags = ITEM_SLOT_SUITSTORE
+	force = 7
 	var/force_on = 57
-	w_class = WEIGHT_CLASS_HUGE
-	throwforce = 10
+	var/force_off = 7
+	wound_bonus = 25
 	throw_speed = 2
 	throw_range = 2
-	wound_bonus = 25
-//	custom_materials = list(MAT_METAL=13000)
-	attack_verb = list("sawn", "torn", "carved", "chopped", "ripped")
-	hitsound = "swing_hit"
-	sharpness = SHARP_EDGED
-	actions_types = list(/datum/action/item_action/startchainsaw)
-	tool_behaviour = TOOL_SAW
-	toolspeed = 0.5
+	throwforce = 10
 	var/on = FALSE
+	tool_behaviour = TOOL_SAW
+	sharpness = SHARP_EDGED
+	toolspeed = 0.3 //slower than a real saw
+	resistance_flags = FIRE_PROOF
+	hitsound = 'sound/weapons/chainsawhit.ogg'
+	var/on_sound = 'sound/weapons/chainsawhit.ogg'
+
+// Description for when turning the ripper on
+/obj/item/twohanded/chainsaw/proc/get_on_description()
+	. = list()
+	.["local_on"] = "<span class ='warning'>You pull the cord, starting up the chainsaw with a roar and letting the blades spin up.</span>"
+	.["local_off"] = "<span class ='notice'>You press the off button, stopping the noise and the carnage.</span>"
+	return
```

Comment:
Is this proc meant to be redefined?
Add these two as object variables instead.
Something like this:
```dm
var/description_on = "<span class ='warning'>You pull the cord, starting up the chainsaw with a roar and letting the blades spin up.</span>"
var/description_off = "<span class ='notice'>You press the off button, stopping the noise and the carnage.</span>"
```

Reply:
This is the only one I could not figure out, its the same code used on the ripper and that was cobbled together from something else, I can't quite remember what. It works though, and I haven't been able how to make it work with the vars you posted,